HACKING.md

```diff
@@ -14,3 +16,106 @@ echo "Ending pre-commit"
 ````
 
 into the `.git/hooks/pre-commit` file.
+
+#### Makefile
+
+* After making a change in the code run `$ make agda_apia_changed`
+* Clean and install Apia. `$ make install-bin`
+* Run Hlint. `$ make hlint`
+* Check the white spaces. `$ make check-whitespace`
+* Generate all. `$ make generated_all`
+* Errors. `$ make errors`
+* Documentation. `$ make haddock`
+* Clean. `$make clean`
+
+
+#### Code Style
+
+* Run always `$ make hlint`
+* Delete *trailing spaces*
+* Use *spaces* instead of tabs
+* Use *two spaces* per each indentation.
+* Each line can not exceed *80 characters.*
+* A file must end by a *newline*.
+* The imports must be *sorted* alphabetically.
+* Try to avoid the *parenthesis* when it is possible.
+  * Use the dollar symbol **$** to avoid redundant parenthesis.
+    Instead of
+
+    ```Haskell
+    let result = f ( g (x `div` y) )
+    ```
+
+    Why not?
+
+    ```Haskell
+    let result = f $ g $ x `div` y
+    ```
+
+    Or better with a composition
+
+    ```Haskell
+    let result = f . g $ x `div` y
+    ```
+
+  * Avoid redundant parenthesis in a *case of* structure.
+
+    Instead of
+
+    ```Haskell
+    case maybeValue of
+        (Just x)→ somethingCool x
+        (Nothing) → failure
+    ```
+
+    It looks nicer presented like
+
+    ```Haskell
+    case maybeValue of
+      Just x   → somethingCool x
+      Nothing  → failure
+    ```
+    Notice in the above example that the arrows are aligned,
+    the indentation follows the two spaces rule, and we remove the parenthesis.
+  * The *arrows* must be *aligned* in the **case of** structure.
+  * Be *explicit* with the types.
+  * In the *let* definition don't forget the *type*, and preserve the structure as follows.
+    Instead of
+
+    ```Haskell
+    let dotApia = homeDir </> ".apia"
+    ```
+
+    do better like
+
+    ```Haskell
+    let dotApia ∷ FilePath
+          dotApia = homeDir </> ".apia"
+    ```
+
+    Or the same rule for the scope of *where*.
+
+    ```Haskell
+    getDefaults ∷ IO Options
+    getDefaults = return somethingNice
+      where
+        somethingNice = defaultsOpts
+    ```
```

Comment:
I asked because Apia uses the `-Wmissing-local-signatures` option. I realised this option doesn't apply in the above example.